File: ports/1.21.1-full/src/main/java/mods/eln/node/NodeManagerNbt.java
package mods.eln.node;

import net.minecraft.nbt.CompoundTag;
import net.minecraft.world.level.saveddata.SavedData;

public class NodeManagerNbt extends SavedData {
    public NodeManagerNbt(String par1Str) {
        super(par1Str);
    }

    @Override
    public boolean isDirty() {
        return true;
    }

    @Override
    public void readFromNBT(CompoundTag nbt) {
        NodeManager.instance.loadFromNbt(nbt);
    }

    @Override
    public CompoundTag writeToNBT(CompoundTag nbt) {
        NodeManager.instance.saveToNbt(nbt, Integer.MIN_VALUE);
        return nbt;
    }
}

File: src/main/kotlin/mods/eln/node/NodeManagerNbt.kt
package mods.eln.node

import net.minecraft.nbt.CompoundTag
import net.minecraft.world.level.saveddata.SavedData
import mods.eln.misc.writeToNBT

class NodeManagerNbt(par1Str: String?) : SavedData(par1Str) {
    override fun isDirty(): Boolean {
        return true
    }

    override fun readFromNBT(nbt: CompoundTag) {
        NodeManager.instance!!.loadFromNbt(nbt)
    }

    override fun writeToNBT(nbt: CompoundTag): CompoundTag {
        //NodeManager.instance.saveToNbt(nbt, Integer.MIN_VALUE);
        return nbt
    }
}
